### DISCUSSION TITLE
I have an issue with federated types

### DISCUSSION THREAD

**Discussion by SamuelAsare** [mirumee/ariadne#997] [Q&A]
Federated fields are not communicating . Any help?

**Comment by rafalp** [comment]
Please show minimal reproduction for the problem or describe how to reproduce it.

**Comment by SamuelAsare** [comment]
I am working on a Timelogs project. The application TIMELOGS is supposed to capture user’s reporting and leaving time. I have two different databases with two different services. The authentication service and the time service. They have different graphql schemas too and are federeated. The reference resolver is been written for the authentication service but when a query is written, the times service is unable to pull the information from the authentication database.

Authentication service
type User @key(fields: "id") {
    id: ID!
    username : String
    first_name: String!
    last_name: String!
    other_name: String
    email: EmailAddress!
    password: String!
    department: String!
    created_at: DateTime
    deleted: Boolean
    deleted_at: DateTime
    role: Role!
}

Time service
type User @key (fields: "id") @extends {
    id : ID! @external
}

type Log @key(fields:"id") {
    id : ID!
    time_in : String
    time_out : String 
    indicator : String
    absent : Boolean!
    user: User @provides (fields:"id")
    user_id: ID
}

@user.reference_resolver
async def resolve_user_reference(_, info, id):
    try:
        await prisma.connect()
        if id:
            return await prisma.user.find_unique(
                where={
                    'id' : int(id)
                }
            )
    except Exception as error:
        raise error
    finally:
        await prisma.disconnect()

**Reply by rafalp** [reply]
It would greatly improve clarity of your posts if you've formatted the code blocks in them. It took me a while to realize what's what in that message. :)

**Reply by SamuelAsare** [reply]
Well noted. Thanks

**Comment by rafalp** [comment]
The problem you are describing, that Service A has no access to data in Service B and viceversa, is not an issue in Ariadne or even in Federation.

Data access and combination is done in federation, and services don't have access to each other data **unless you explicitly implement that access into services**. Time service will not know about the authentication service by default. They will only know about user ID. If you want to know more, your time service should use this ID to request data from authentication service directly, either by doing GraphQL query to it, or asking it's REST API or by other method.

What Federation does when it finds that `user` is queried for on type `Log`, it takes user ID from data returned by `user` resolver in Time service, and then queries special field in Authentication service named `_Entity` for requested user fields for user with ID returned from "Time" service.

Keeping above in mind, Time service should know atleast about user's ID in Authentication service, and `resolve_user_reference` in Time service should return `{"id": "USER_ID_STORED_IN_TIME_SERVICE"}`, and leave the rest to Federation.

**Comment by SamuelAsare** [comment]
I have done a similar solution you've presented but still does not generate
the data.

Is there a way to connect with you to have a look at my code? Video
conference or share my code with you on github.

**Reply by rafalp** [reply]
Sorry, this is an open source project - Your best bet is to put code demonstrating the issue somewhere on github for Us to look at.

**Reply by SamuelAsare** [reply]
https://github.com/Samuelasare85/timelogs-times-services.git
https://github.com/Samuelasare85/timelogs-authentication-services.git

The resolvers are in the src folder, and it's linked to the main.py

**Reply by rafalp** [reply]
Ok, thanks. What happens in `resolve_user_reference`? Is it called for federated query? If it is, what gets passed to the `id` argument?

I'm sorry I'm unable to give this a closer look at the moment - I'm on vacation and will return to the office in january 2023.
